src/components/tx/AdvancedParamsForm/index.tsx

```diff
@@ -31,13 +34,14 @@ enum AdvancedField {
 
 export type AdvancedParameters = {
   [AdvancedField.nonce]: number
+  [AdvancedField.userNonce]: number
   [AdvancedField.gasLimit]?: BigNumber
   [AdvancedField.maxFeePerGas]?: BigNumber
   [AdvancedField.maxPriorityFeePerGas]?: BigNumber
   [AdvancedField.safeTxGas]?: number
 }
 
-type AdvancedParamsFormProps = AdvancedParameters & {
+type AdvancedParamsFormProps = Omit<AdvancedParameters, AdvancedField.userNonce> & {
```

Comment:
Why are we not passing the `userNonce` here like we do for the other values? It is currently not persisted when editing the field.

Reply:
When you merge your refactoring, I'll move useUserNonce to the new component, so it will be passed to the form as a prop.